pkg/kapp/logs/view.go

```diff
@@ -66,6 +68,8 @@ func (v View) Show(cancelCh chan struct{}) error {
 				return fmt.Sprintf("%s > %s", pod.Name, cont.Name)
 			}
 
+			v.tailOpts.ContainerNames = v.contFilterFunc(pod)
```

Comment:
im not certain how go works this out with go func()... "modifying" concurrently. let's change it so that we copy tailOpts to be more explicit:

```
tailOpts := v.tailOpts
tailOpts.ContainerNames = v.contFilterFunc(pod)
```

and use it below in NewPodLog

Reply:
Agree.